src/Caching/StackExchangeRedis/src/RedisOutputCacheStore.cs

```diff
@@ -0,0 +1,487 @@
+
+        try
+        {
+            return (byte[]?)(await cache.StringGetAsync(GetValueKey(key)).ConfigureAwait(false));
+        }
+        catch (Exception ex)
+        {
+            OnRedisError(ex, cache);
+            throw;
+        }
+    }
+
+    async ValueTask<bool> IOutputCacheBufferWriterStore.GetAsync(string key, IBufferWriter<byte> destination, CancellationToken cancellationToken)
+    {
+        ArgumentNullThrowHelper.ThrowIfNull(key);
+        ArgumentNullThrowHelper.ThrowIfNull(destination);
+
+        var cache = await ConnectAsync(cancellationToken).ConfigureAwait(false);
+        Debug.Assert(cache is not null);
+
+        Lease<byte>? result = null;
+        try
+        {
+            result = await cache.StringGetLeaseAsync(GetValueKey(key)).ConfigureAwait(false);
+            if (result is null)
+            {
+                return false;
+            }
+
+            destination.Write(result.Span);
+            return true;
+        }
+        catch (Exception ex)
+        {
+            result?.Dispose();
+            OnRedisError(ex, cache);
+            throw;
+        }
+    }
+
+    ValueTask IOutputCacheStore.SetAsync(string key, byte[] value, string[]? tags, TimeSpan validFor, CancellationToken cancellationToken)
+    {
+        ArgumentNullException.ThrowIfNull(value);
+        return ((IOutputCacheStore)this).SetAsync(key, new ReadOnlySequence<byte>(value), tags.AsMemory(), validFor, cancellationToken);
+    }
+
+    async ValueTask IOutputCacheStore.SetAsync(string key, ReadOnlySequence<byte> value, ReadOnlyMemory<string> tags, TimeSpan validFor, CancellationToken cancellationToken)
+    {
+        var cache = await ConnectAsync(cancellationToken).ConfigureAwait(false);
+        Debug.Assert(cache is not null);
+
+        byte[]? leased = null;
+        ReadOnlyMemory<byte> singleChunk;
+        if (value.IsSingleSegment)
+        {
+            singleChunk = value.First;
+        }
+        else
+        {
+            int len = checked((int)value.Length);
+            leased = ArrayPool<byte>.Shared.Rent(len);
+            value.CopyTo(leased);
+            singleChunk = new(leased, 0, len);
+        }
+
+        await cache.StringSetAsync(GetValueKey(key), singleChunk, validFor).ConfigureAwait(false);
+        // only return lease on success
+        if (leased is not null)
+        {
+            ArrayPool<byte>.Shared.Return(leased);
+        }
+
+        if (!tags.IsEmpty)
+        {
+            long expiryTimestamp = GetExpirationTimestamp(validFor);
+            var len = tags.Length;
+
+            // tags are secondary; to avoid latency costs, we'll use fire-and-forget when adding tags - this does
+            // mean that in theory tag-related error may go undetected, but: this is an acceptable trade-off
+            const CommandFlags TagCommandFlags = CommandFlags.FireAndForget;
+
+            for (int i = 0; i < len; i++) // can't use span in async method, so: eat a little overhead here
```

Comment:
Nit:
```suggestion
            for (var i = 0; i < len; i++) // can't use span in async method, so: eat a little overhead here
```
Use `var` per convention in this repo.